AdobeStockImage/Model/GetSavedImageFilePath.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\AdobeStockImage\Model;
+
+use Magento\AdobeStockImage\Model\Storage\Delete as StorageDelete;
+use Magento\AdobeStockImage\Model\Storage\Save as StorageSave;
+use Magento\Framework\Api\Search\Document;
+use Magento\Framework\Exception\AlreadyExistsException;
+use Magento\Framework\Exception\CouldNotDeleteException;
+use Magento\Framework\Exception\FileSystemException;
+
+/**
+ * Save asset file and retrieve its path.
+ */
+class GetSavedImageFilePath implements GetSavedImageFilePathInterface
```

Comment:
@sivaschenko I changed the `SaveImageFile` name to this one based on the discussion with @lenaorobei 
![image](https://user-images.githubusercontent.com/5318512/78451678-b8242980-768f-11ea-9f5c-f13cbba6db28.png)

Reply:
My opinion that `SaveImageFile` good but also it returns the path of a file. As a variant, I used `RetrievePathFromImageSave`. What do you think about this?

Reply:
@lenaorobei I'd keep the return of the `SaveImageFile` in order to prevent discrepancies in case that path is adjusted by the storage. There is no way to retrieve the correct file path later in this case.

Please let me know what you think

Reply:
@coderimus Olena meant that the service should either save the file or provide the file path. This particular service is used to save the file, so the naming should not be changed. The question is whether to return the $path

Reply:
@sivaschenko thank you for details. I will check this. Also, as I understood, we need to return the name of the SaveMediaGalleryAsset, right? But in that case, we have to return id of the asset.

Reply:
If the service can be used each time I need to retrieve the path - "RetrievePath..." naming should be ok.

If the service is primarily used to save the file - it should be named accordingly.

`SaveImageFile` sounds good to me

Reply:
@coderimus after conversation with @lenaorobei we come to the next conclusions:
- The class name should be `SaveImageFile`
- The execute method should not return the path (use the argument destinationPath in clients instead) - return type `void`
- Update `\Magento\AdobeStockImage\Model\Storage\Save::execute` to make destinationPath a required parameter and chance the return type to `void`